### ISSUE TITLE
Use base aws classes in amazon provider Operators/Sensors/Triggers

### ISSUE THREAD

**Issue by Taragolis** [apache/airflow#35278]
### Body

Right now some inconsistency exists between different Operators/Sensors/Triggers which are use `AwsBaseGeneric` hook.
After #34784 and #34884 we could make all of them (except Transfer Operators) use same subset of generic attributes.

It might make more easier to end users use this objects (don't need to think is it `region` or `region_name` or is it even support this arguments), as well as make easier extend by new new hook parameters to all this operators in the one go.

General Approach for solve each case:
- Use `AwsBaseOperator` for operators, `AwsBaseSensor ` for sensors. In some very specific case e.g. multi inheritance required to use `AwsBaseHookMixin` (see: [AppflowRecordsShortCircuitOperator](https://github.com/apache/airflow/blob/bc4a22c6bd8096e7b62147031035cb14896fe934/airflow/providers/amazon/aws/operators/appflow.py#L399))
- Provide additional arguments to `AwsBaseWaiterTrigger` (if required)
- Wrap template_fields parameters 
- Attach generic documentation into the service specific page into the [Amazon Provider Documentation](https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/operators/index.html)
- Validate that all parameters propagated to operator/sensor specific hook 

All this steps already done for some of the operators, so it  could be use as a reference

- [x] Use base aws classes in Amazon AppFlow Operators (https://github.com/apache/airflow/pull/35082)
    + [operator(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/operators/appflow.py)
 - [x] Use base aws classes in Amazon Athena Operators/Sensors/Triggers (https://github.com/apache/airflow/pull/35133)
    + [operator(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/operators/athena.py)
    + [sensor(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/sensors/athena.py)
    + [trigger(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/triggers/athena.py)
 - [ ] Use base aws classes in AWS Batch Operators/Sensors/Triggers (https://github.com/apache/airflow/pull/35226) 
    + [operator(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/operators/batch.py) https://github.com/apache/airflow/pull/49172
    + [sensor(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/sensors/batch.py) https://github.com/apache/airflow/pull/49172
    + [trigger(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/triggers/batch.py)
 - [x] Use base aws classes in AWS CloudFormation Operators/Sensors https://github.com/apache/airflow/pull/36771
    + [operator(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/operators/cloud_formation.py)
    + [sensor(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/sensors/cloud_formation.py)
 - [x] Use base aws classes in AWS Datasync Operators https://github.com/apache/airflow/pull/36766
    + [operator(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/operators/datasync.py)
 - [x] Use base aws classes in AWS DMS Operators/Sensors https://github.com/apache/airflow/pull/36772
    + [operator(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/operators/dms.py)
    + [sensor(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/sensors/dms.py)
 - [x] Use base aws classes in Amazon DynamoDB Sensors https://github.com/apache/airflow/pull/36770
    + [sensor(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/sensors/dynamodb.py)
 - [x] Use base aws classes in Amazon EC2 Operators/Sensors/Triggers https://github.com/apache/airflow/pull/47931
    + [operator(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/operators/ec2.py)
    + [sensor(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/sensors/ec2.py)
    + [trigger(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/triggers/ec2.py)
 - [x] Use base aws classes in Amazon ECS Operators/Sensors/Triggers (https://github.com/apache/airflow/pull/36393)
    + [operator(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/operators/ecs.py)
    + [sensor(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/sensors/ecs.py)
    + [trigger(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/triggers/ecs.py)
 - [x] Use base aws classes in Amazon EKS Operators/Sensors/Triggers https://github.com/apache/airflow/pull/48192
    + [operator(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/operators/eks.py)
    + [sensor(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/sensors/eks.py)
    + [trigger(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/triggers/eks.py)
 - [x] Use base aws classes in Amazon EMR Operators/Sensors/Triggers https://github.com/apache/airflow/pull/49486
    + [operator(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/operators/emr.py)
    + [sensor(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/sensors/emr.py)
    + [trigger(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/triggers/emr.py)
 - [x] Use base aws classes in Amazon EventBridge Operators https://github.com/apache/airflow/pull/36765
    + [operator(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/operators/eventbridge.py)
 - [x] Use base aws classes in AWS Glue Operators/Sensors/Triggers 
    + [operator(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/operators/glue.py) https://github.com/apache/airflow/pull/49750
    + [sensor(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/sensors/glue.py) https://github.com/apache/airflow/pull/52243
    + [trigger(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/triggers/glue.py) https://github.com/apache/airflow/pull/52243
 - [x] Use base aws classes in AWS Glue Data Catalog Sensors https://github.com/apache/airflow/pull/40492
    + [sensor(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/sensors/glue_catalog_partition.py)
 - [x] Use base aws classes in AWS Glue Crawlers Operators/Sensors/Triggers https://github.com/apache/airflow/pull/40504
    + [operator(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/operators/glue_crawler.py)
    + [sensor(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/sensors/glue_crawler.py)
    + [trigger(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/triggers/glue_crawler.py) 
 - [x] Use base aws classes in AWS Glue DataBrew Operators/Triggers https://github.com/apache/airflow/pull/41848
    + [operator(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/operators/glue_databrew.py)
    + [trigger(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/triggers/glue_databrew.py)
 - [x] Use base aws classes in AWS Lambda Operators/Sensors/Triggers (https://github.com/apache/airflow/pull/34890)
    + [operator(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/operators/lambda_function.py)
    + [sensor(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/sensors/lambda_function.py)
    + [trigger(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/triggers/lambda_function.py)
 - [x] Use base aws classes in Amazon QuickSight Operators/Sensors https://github.com/apache/airflow/pull/36776
    + [operator(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/operators/quicksight.py)
    + [sensor(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/sensors/quicksight.py)
 - [x] Use base aws classes in Amazon RDS Operators/Sensors/Triggers https://github.com/apache/airflow/pull/48872
    + [operator(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/operators/rds.py)
    + [sensor(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/sensors/rds.py)
    + [trigger(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/triggers/rds.py)
 - [ ] Use base aws classes in Amazon Redshift clusters Operators/Sensors/Triggers
    + [operator(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/operators/redshift_cluster.py) https://github.com/apache/airflow/pull/51129
    + [sensor(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/sensors/redshift_cluster.py) https://github.com/apache/airflow/pull/51129
    + [trigger(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/triggers/redshift_cluster.py)
 - [x] Use base aws classes in AWS Redshift Data API Operators https://github.com/apache/airflow/pull/36764
    + [operator(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/operators/redshift_data.py)
 - [x] Use base aws classes in Amazon S3 Operators/Sensors/Triggers https://github.com/apache/airflow/pull/47321
    + [operator(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/operators/s3.py)
    + [sensor(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/sensors/s3.py)
    + [trigger(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/triggers/s3.py)
 - [x] Use base aws classes in Amazon S3 Glacier Operators/Sensors (https://github.com/apache/airflow/pull/35108)
    + [operator(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/operators/glacier.py)
    + [sensor(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/sensors/glacier.py)
 - [ ] Use base aws classes in Amazon SageMaker Operators/Sensors/Triggers 
    + [operator(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/operators/sagemaker.py) https://github.com/apache/airflow/pull/50321
    + [sensor(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/sensors/sagemaker.py) https://github.com/apache/airflow/pull/50321
    + [trigger(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/triggers/sagemaker.py)
 - [x] Use base aws classes in Amazon SNS Operators https://github.com/apache/airflow/pull/36615
    + [operator(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/operators/sns.py)
 - [x] Use base aws classes in Amazon SQS Operators/Sensors/Triggers https://github.com/apache/airflow/pull/36613
    + [operator(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/operators/sqs.py)
    + [sensor(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/sensors/sqs.py)
    + [trigger(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/triggers/sqs.py)
 - [x] Use base aws classes in AWS Step Functions Operators/Sensors/Triggers https://github.com/apache/airflow/pull/36468
    + [operator(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/operators/step_function.py) 
    + [sensor(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/sensors/step_function.py)
    + [trigger(s)](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/triggers/step_function.py)

### Committer

- [X] I acknowledge that I am a maintainer/committer of the Apache Airflow project.

**Comment by ferruzzi** [comment]
Thanks for getting this set up!

**Comment by gopidesupavan** [comment]
@Taragolis i am happy to hook into this `Use base aws classes in AWS Glue Crawlers Operators/Sensors/Triggers` ? or anyone is working already?

**Comment by prakharcode** [comment]
@Taragolis I am happy to take EMR, EC2 and S3 (operators) here

**Comment by gopidesupavan** [comment]
@vincbeck @eladkal Working on this `Use base aws classes in AWS Glue Data Catalog Sensors` didnt find anything in the documentation and example code for this. Planning to add them here https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/operators/glue.html is this fine?

**Comment by vincbeck** [comment]
> @vincbeck @eladkal Working on this `Use base aws classes in AWS Glue Data Catalog Sensors` didnt find anything in the documentation and example code for this. Planning to add them here https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/operators/glue.html is this fine?

Please yes :)

**Comment by eladkal** [comment]
@gopidesupavan do you plan to address the other Glue services that were left?

**Comment by gopidesupavan** [comment]
@eladkal Yes will update other services, coming week.

**Comment by ellisms** [comment]
I'll take a look at EC2.

**Comment by hinchliff** [comment]
I'd love to see the RDS sensors extend from the `AwsBaseSensor`.  We have a use-case where we need to support multiple regions, and requiring a new Connection for each region is painful.

**Comment by dominikhei** [comment]
Glue Databrew was added in #41848 

I'll have a look at the remaining Glue ones.

**Comment by dominikhei** [comment]
> Glue Databrew was added in [#41848](https://github.com/apache/airflow/pull/41848)
> 
> I'll have a look at the remaining Glue ones.

Speaking on this. The `GlueJobCompleteTrigger` currently does not come with a waiter in boto3. @ellisms as you implemented the Operator which pattern do you think is preferable? Is there something I am overlooking.

1. Not letting the Trigger inherit from AwsBaseWaiterTrigger and just port the Sensor. This is like it is done with the [GlueCatalogPartitionSensor](https://github.com/apache/airflow/blob/main/providers/amazon/src/airflow/providers/amazon/aws/sensors/glue_catalog_partition.py#L36) which inherits from the base class, whilst its trigger [does not](https://github.com/apache/airflow/blob/main/providers/amazon/src/airflow/providers/amazon/aws/triggers/glue.py#L78)

 2. Or create a new custom waiter and port the Trigger to the  `AwsBaseWaiterTrigger`?

-----
I have now opted for the second option, will test all changes with dags and then open a Draft PR

**Comment by ellisms** [comment]
@dominikhei I'd go with option 2. A quick scan of the current waiter code in the hook doesn't reveal anything special. A custom waiter would be able to handle it.

**Comment by Cloud-Architect-Emma** [comment]
@eladkal Thanks for the suggestion! I’m happy to start with a smaller task.

I’ve reviewed the remaining items in #35278 and I can take ownership of one of the AWS provider components that still needs to be migrated to the base classes. Unless there’s a preferred priority, I’ll select one of the open items and begin working on a PR following the existing patterns.

Happy to contribute and iterate based on feedback.

**Comment by KarshVashi** [comment]
I'd like to take the remaining trigger migrations for Batch, Redshift cluster and SageMaker if they are still pending. The operators and sensors already seem merged for all three and triggers still need verify and botocore_config propagated through to the hook, following the same pattern as the Lambda trigger.

I'll do these as separate PRs

**Comment by dominikhei** [comment]
> I'd like to take the remaining trigger migrations for Batch, Redshift cluster and SageMaker if they are still pending. The operators and sensors already seem merged for all three and triggers still need verify and botocore_config propagated through to the hook, following the same pattern as the Lambda trigger.
> 
> I'll do these as separate PRs

I wonder whether it would make sense to keep the ClusterTrigger as is, as due to it not checking a single status but multiple ones, one would need to introduce custom waiters for every possible state, which are then used dynamically.

**Comment by KarshVashi** [comment]
Good point, I'll start with the Batch triggers which seem cleanest and then understand the complexity of others

**Comment by KarshVashi** [comment]
For Redshift cluster triggers, five of the six triggers already use `AwsBaseWaiterTrigger` and need the same `verify`/`botocore_config` propagation:
> - `RedshiftCreateClusterTrigger`
> - `RedshiftPauseClusterTrigger`
> - `RedshiftCreateClusterSnapshotTrigger`
> - `RedshiftResumeClusterTrigger`
> - `RedshiftDeleteClusterTrigger`

Would it make sense to implement migration for the above ones and skipping `RedshiftClusterTrigger` as it uses a custom polling loop with `BaseTrigger`?

**Comment by eladkal** [comment]
All items handled
